File: src/com/stephenclhc/ReverseInteger.java
package com.stephenclhc;

public class ReverseInteger {
    // 7. Reverse Integer

    public int reverse(int x) {
        boolean isNegative =  x < 0;

        x = Math.abs(x);

        int result = 0;
        while (x > 0){
            //It is a 32-bit integer. to check if it is larger or smaller than the limits
            if(result > Integer.MAX_VALUE / 10 ||
                    (-result < Integer.MIN_VALUE /10 && isNegative))
                return 0;
            result = result * 10 + x%10;
            x/= 10;
        }

        if (isNegative)
            return  - result;
        else return result;
    }
}

File: Python/Reverse_Integer.py
# 7. Reverse Integer


def reverse(self, x: int) -> int:
    is_negative = x < 0

    x = abs(x)
    reverse_num = 0

    while x > 0:
        reverse_num = reverse_num * 10 + x % 10
        x //= 10

    if reverse_num > pow(2, 31):
        return 0
    return -reverse_num if is_negative else reverse_num
